backend/molgenis-emx2/src/main/java/org/molgenis/emx2/Column.java

```diff
@@ -614,8 +614,13 @@ public boolean isPrimaryKey() {
   }
 
   @Override
-  public int compareTo(Object o) {
-    // position is required in backend
-    return getPosition().compareTo(((Column) o).getPosition());
+  public int compareTo(Column o) {
+    if (this.getPosition() > o.getPosition()) {
+      return 1;
+    } else if (this.getPosition() < o.getPosition()) {
+      return -1;
+    } else {
+      return this.getName().compareTo(o.getName());
+    }
```

Comment:
looks oke , but note that; if there are two column instances c1, c2. And c1.pos === c2.pos and c1.name === c2.name and collection foo  contains c1 and has length n , adding c2 to collection foo , collection foo still has length n. Some other collection bar ( using equals ) could result in length = n + 1 after adding c2. because compareTo and equals are not in sync